### ISSUE TITLE
ApplicationBlueprinter pattern

### ISSUE THREAD

**Issue by toobulkeh** [procore/blueprinter#80]
Is there any reason not to use the `Application` pattern with Blueprinter?

`application_blueprint.rb`:
```ruby
class ApplicationBlueprint < Blueprinter::Base
  identifier :id
end
```

`user_blueprint.rb`:
```ruby
class UserBlueprint < ApplicationBlueprint
  fields :name
end
```

**Comment by philipqnguyen** [comment]
Definitely! There is no reason not to use the `ApplicationBlueprint` pattern. It is already a pattern that Procore is using internally as well and I highly recommend using that pattern in your projects.

**Comment by toobulkeh** [comment]
Apparently extending a blueprint with an identifier removes it from the returned value. Is that expected?

IE, I had to add this line back to all of our blueprints:

```ruby
identifier :id
```

**Comment by philipqnguyen** [comment]
I did a short test of your example locally and I can confirm you're correct and that it does not work. Will  have to look into it further.

**Comment by Systho** [comment]
The problem is the view collection is a class instance variable, thus it's not the same in the parent class and in the subclass.

I've personally fixed the problem with this snippet in my BaseBlueprint

```
  class BaseBlueprint < Blueprinter::Base

    def self.inherited(sub_class)
      sub_class.identifier :id
    end

end
```

**Comment by philipqnguyen** [comment]
@Systho you're right, class instance variable is the problem. I think you have the right solution.

**Comment by philipqnguyen** [comment]
resolved by #105
